### PR TITLE
allow calling index.add, index.move and index.remove with single items

### LINKED ISSUE #813: repo.index.add -> check if input is a list
For ~~stupid~~ people like me:
Could you please add a check if the input to `repo.index.add(...)` is a list, and not a string?
It took me a while to find out that this was the real error behind:

> ValueError: Absolute path '\\' is not in git repository at 'C:\\Users\\xxx\\yyy\\zzz'

:-)

### PR DESCRIPTION
**Pull request by MarcelBeining**
added testing for it
closes #813

### CODE REVIEW

**Comment by MarcelBeining** [comment]
hm guys, I am not that into testing to understand why the tests failed in some environments, maybebe you can help me. =/ thanks

The only thing I see is that in AppVeyor, the test fails when I try to remove the added file again from the index with  
`stderr: 'fatal: pathspec 'git\test\test_index.py' did not match any files`
however why is this file there in some environments but not there in others?

File: git/index/base.py [hunk 1]
```diff
@@ -569,16 +569,23 @@ def _preprocess_add_items(self, items):
...
-            if isinstance(item, string_types):
-                paths.append(self._to_relative_path(item))
-            elif isinstance(item, (Blob, Submodule)):
-                entries.append(BaseIndexEntry.from_blob(item))
-            elif isinstance(item, BaseIndexEntry):
-                entries.append(item)
-            else:
-                raise TypeError("Invalid Type: %r" % item)
+        
+        if isinstance(items, string_types):
```
Comment (Byron) [inline git/index/base.py]: I think the repetition can and should be removed. My suggestion is to check if the input type is something you can iterate, and if not, pack it into a list. The rest of the code would then remain unchanged.
Does that make sense?

File: git/index/base.py [hunk 2]
```diff
@@ -801,13 +808,18 @@ def _items_to_rela_paths(self, items):
...
         may be absolute or relative paths, entries or blobs"""
         paths = []
-        for item in items:
-            if isinstance(item, (BaseIndexEntry, (Blob, Submodule))):
-                paths.append(self._to_relative_path(item.path))
-            elif isinstance(item, string_types):
-                paths.append(self._to_relative_path(item))
-            else:
-                raise TypeError("Invalid item type: %r" % item)
+        if isinstance(items, (BaseIndexEntry, (Blob, Submodule))):
```
Comment (Byron) [inline git/index/base.py]: I think alternative implementation suggested above should also be applied here.

**Comment by Byron** [comment]
Thanks a lot for your contribution thus far !
I am closing this PR due to prolonged inactivity. Please feel free to reopen it in case it can proceed.

**Pull request closed without merging**